src/Compilers/CSharp/Portable/Symbols/MethodSymbolExtensions.cs

```diff
@@ -189,6 +190,11 @@ public static bool IsIAsyncEnumeratorReturningAsync(this MethodSymbol method, CS
 
         internal static CSharpSyntaxNode ExtractReturnTypeSyntax(this MethodSymbol method)
         {
+            if (method is SynthesizedSimpleProgramEntryPointSymbol synthesized)
```

Comment:
> I wouldn't expect this to work at all. This method is asking for the syntax of the stated return type, but we don't have a stated return type.

We need this method to return reasonable non-null node. To the user it can be surfaced only as an error location. The current implementation is reasonable enough, in my opinion. There is a PROTOTYPE comment to give it another thought later.

---
In reply to: [382794614](https://github.com/dotnet/roslyn/pull/41706#discussion_r382794614) [](ancestors = 382794614)